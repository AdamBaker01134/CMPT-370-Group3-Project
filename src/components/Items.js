import React from "react";
import PropTypes from "prop-types";

function Items(props) {
    const { items = [] } = props;

    // single item, render a span
    if (items.length === 1) {
        return <span>{items[0]}</span>;
    } 

    // multiple items, render a list
    if (items.length > 1) {
        return (
            <ul>
                {items.map(item => <li key={item}>{item}</li>)}
            </ul>
        );
    }

    // no items, render a message
    return <span>No items in list</span>;
}

Items.propTypes = {
    items : PropTypes.array,
};

Items.defaultProps = {
    items: []
};

export { Items };