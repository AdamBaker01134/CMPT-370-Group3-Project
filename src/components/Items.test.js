import React from 'react';
import renderer from 'react-test-renderer';

import { Items } from './Items';

describe("snapshot tests for a simple react program", () => {
    it('renders correctly where there are no items', () => {
        const tree = renderer.create(<Items />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    it('renders correctly where there is a single item', () => {
        const items = ['one'];
        const tree = renderer.create(<Items items={items} />).toJSON();
        expect(tree).toMatchSnapshot();
    })
    
    it('renders correctly where there are multiple items', () => {
        const items = ['one', 'two', 'three'];
        const tree = renderer.create(<Items items={items} />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})