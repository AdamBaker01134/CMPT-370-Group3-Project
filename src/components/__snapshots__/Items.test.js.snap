// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot tests for a simple react program renders correctly where there are multiple items 1`] = `
<ul>
  <li>
    one
  </li>
  <li>
    two
  </li>
  <li>
    three
  </li>
</ul>
`;

exports[`snapshot tests for a simple react program renders correctly where there are no items 1`] = `
<span>
  No items in list
</span>
`;

exports[`snapshot tests for a simple react program renders correctly where there is a single item 1`] = `
<span>
  one
</span>
`;
